hibike/serial_enumeration_test.py

```diff
@@ -0,0 +1,28 @@
+import threading
+import hibike_message as hm
+import serial
+import glob
+
+# get all available hid devices
+devices = [serial.Serial(port) for port in glob.glob("/dev/ttyACM*") + glob.glob("/dev/ttyUSB*") + glob.glob("/dev/tty.usbmodem*")]
+
+def print_uid(i, device):
+    for packet in hm.blocking_read_generator(device):
+        if packet.getmessageID() == hm.messageTypes["SubscriptionResponse"]:
+            params, delay, uid = hm.parse_subscription_response(packet)
+            print("thread %d enumerated %s with uid %d" % (i, hm.uid_to_device_name(uid), uid))
+            return
+
+# start up a read thread for each device
+read_threads = [threading.Thread(target=print_uid, args=(i, device)) for (i, device) in enumerate(devices)]
+
+for read_thread in read_threads:
+    read_thread.start()
+
+# ping each device
+for device in devices:
+    hm.send(device, hm.make_ping())
+
+# wait until each thread enumerates its device and exits
```

Comment:
the read threads exit as soon as they successfully enumerate their device. The joins are so the test can exit if it enumerates successfully on every port